src/Types/Expiry.cs

```diff
@@ -0,0 +1,140 @@
+using System.Globalization;
+using Concordium.Sdk.Helpers;
+
+namespace Concordium.Sdk.Types;
+
+/// <summary>
+/// Represents an expiration time for a transaction.
+///
+/// A transaction can not be included in a block if its expiration time is before the block (slot) time.
+/// </summary>
+public readonly struct Expiry : IEquatable<Expiry>
+{
+    public const int BytesLength = sizeof(ulong);
+
+    /// <summary>
+    /// Time at which the transaction expires.
+    /// </summary>
+    private readonly DateTimeOffset _timestamp;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Expiry"/> class.
+    /// </summary>
+    /// <param name="timestamp">Time at which the transaction expires.</param>
+    private Expiry(DateTimeOffset timestamp) => this._timestamp = timestamp;
+
+    /// <summary>
+    /// Creates a new expiration time that is <paramref name="minutes"/> ahead of the current system time.
+    /// </summary>
+    /// <param name="minutes">An expiration time specified by the number of minutes from the current system time.</param>
+    public static Expiry AtMinutesFromNow(ulong minutes) => From(Now().AddMinutes(minutes));
+
+    /// <summary>
+    /// Creates a new expiration time that is <paramref name="seconds"/> from the current system time.
+    /// </summary>
+    /// <param name="seconds">An expiration time specified by the number of seconds from the current system time.</param>
+    public static Expiry AtSecondsFromNow(ulong seconds) => From(Now().AddSeconds(seconds));
+
+    /// <summary>
+    /// Creates an instance from a expiration time represented by the elapsed number of seconds since the UNIX epoch.
+    /// </summary>
+    /// <param name="secondsSinceEpoch">Expiration time represented by the elapsed seconds since the UNIX epoch.</param>
+    /// <exception cref="ArgumentOutOfRangeException">The supplied <see cref="ulong"/> timestamp has a value which exceeds the maximum value of a <see cref="long"/>.</exception>
+    public static Expiry From(ulong secondsSinceEpoch)
+    {
+        // We check due to the type of FromUnixTimeSeconds.
+        if (secondsSinceEpoch > long.MaxValue)
+        {
+            throw new ArgumentOutOfRangeException(
+                $"The supplied ulong timestamp has a value of {secondsSinceEpoch} which exceeds the maximum value of long."
+            );
+        }
+        return new Expiry(DateTimeOffset.FromUnixTimeSeconds((long)secondsSinceEpoch));
+    }
+
+    /// <summary>
+    /// Creates an instance whose expiration time is the current system time.
+    /// </summary>
+    /// <exception cref="ArgumentOutOfRangeException">System time as number of seconds since UNIX epoch should be larger than 0.</exception>
+    public static DateTimeOffset Now()
```

Comment:
Do we needs this function? It just calls DateTimeOffset.UtcNow - couldn’t the user just use this?

Just a question fair enough if we want to be explicit in some way.